src/FileSystem-Disk/DiskStore.class.st

```diff
@@ -79,12 +85,46 @@ DiskStore class >> maxFileNameLength [
 	self subclassResponsibility 
 ]
 
+{ #category : #public }
+DiskStore class >> readOnlyVariant [
+
+	"return a class variant for read-only store. It should be properly defined in my subclasses"
+
+	^ self
```

Comment:
Shouldn't it be a subclass responsibility or an error `NoReadOnlyStoreAvailableForPlatform`?

Reply:
I was thinking about subclass responsibility but I wanted to keep compatibility with possible external subclasses.

Reply:
Well, I don't think it will break compatibility because if someone has his own store, then the code to use it will not use your read-only option. 

Then, if the user wants to use the read-only option, he should be notified that his store is not able to do that.